### ISSUE TITLE
Missing documentation on `$` prefix for parameters

### ISSUE THREAD

**Issue by chuanqisun** [cozodb/cozo#135]
Hi, I encountered some examples of `$` parameters on these pages:

- `$AIR_ROUTES_NODES_URL` in https://docs.cozodb.org/en/latest/tutorial.html#Importing-dataset-the-hard-way
- `$q` https://docs.cozodb.org/en/latest/vector.html

The problem is that there is no explicit documentation for such parameter syntax in CozoScript. The two occurrences above appear in advanced topics so new users might not find them easily.

Some ideas:

- Add a section on [Queries](https://docs.cozodb.org/en/latest/queries.html#) page about this syntax
- Document parameter binding in one of the host languages (maybe Rust?) and point other languages to that location
  - SQLite, for example, documents the parameter syntax in the [C-language interface](https://www.sqlite.org/c3ref/bind_blob.html)

**Comment by chuanqisun** [comment]
As a related issue, search `$` on the documentation sidebar yield no results. It was probably escaped/sanitized.

![image](https://github.com/cozodb/cozo/assets/1895289/73893f33-845a-4a0f-b0b7-0e416311433b)

**Comment by chuanqisun** [comment]
Quick investigation: Sphinx's indexer seems to strip `$` from the terms. Here is the raw index: https://docs.cozodb.org/en/latest/searchindex.js

Looking for `AIR_ROUTES_NODES_URL` in that file, we can confirm that the `$` is not part of the "term"
![image](https://github.com/cozodb/cozo/assets/1895289/67eb0b31-327d-477d-bb4c-5ed29732df01)

I wasn't able to dig further. I think it's either an `nbsphinx` [parser](https://github.com/spatialaudio/nbsphinx/blob/master/src/nbsphinx/__init__.py) issue, or an Sphinx indexer issue. It was just a nitpick btw.

**Comment by gisborne** [comment]
I’d be delighted to help with that.

I’m an experienced technical writer and developer, with a strong knowledge of formal logic and the relational model.

**Comment by zh217** [comment]
@gisborne That would be very much appreciated! I think it would be best to proceed in this way:

- In the future, there will be alpha and beta releases
  - These will be source code-only, as building all the artefacts is quite time- and resource-intensive
- In the release notes, new features will be very briefly explained
- Anyone at any time can submit pull requests to https://github.com/cozodb/cozo-docs/tree/dev for changes in the documentation, not only restricted to documenting new features
- Before each final release, the changes will be consolidated so as to make sure that the documentation stay coherent and consistent with the features

**Comment by gisborne** [comment]
Great!

Is there anything you want to update now, or shall we wait on the new release (rough ETA?).

Some comments you’ve made lead me to believe you have a rough roadmap in mind. Even a rough one would be good to share, if I can help with that.

I love everything about this project. My personal interest is in getting the industry to switch from SQL to Datalog, and to shift most business logic into Datalog. I think there’s a potential revolution in engineering efficiency to be had there. I’m hoping before long that Cozo will be a serious industrial relational database, with failover and scalability. Various things you’ve said lead me to understand that you are moving in that direction.

FWIW.

**Comment by zh217** [comment]
Our development is currently guided by the needs of our other projects, mainly focused on facilitating and automating scientific and social research. The recent features show how Language Learning Models (LLMs) have influenced these projects. Therefore, we're concentrating on improving text processing with complex logic. We're not heavily focused on big data or distributed processing at this time. While we don't prioritize speed, CozoDB, being built on Rust, is quite fast.

CozoDB is actively used in these projects in production, which means we might not have enough resources to meet all the demands of the open-source community. However, this also allows us to develop an authentically open-source CozoDB without relying on venture capital, which can sometimes harm open-source projects in the long run. As an open-source project, we invite anyone who wants to contribute new features or even fork the project.

**Comment by gisborne** [comment]
Are you saying by this that it isn’t on the roadmap to implement features that make Cozo suitable as a business database (probably failover is what’s needed)?

**Comment by zh217** [comment]
No I'm not saying that.

**Comment by gisborne** [comment]
Delighted to hear.

In the interim, I’m ready to help with documentation as I can. And I’m going to learn Rust to see if I can start contributing to the code.

**Comment by zh217** [comment]
The next release will have to wait for a while, as I have my hands full at the moment.
